src/lib/search-params.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  SearchOptions,
+  SortBy as SortBySearchOption,
+  SortOrder,
+} from '@/lib/dataset-fetcher';
+
+export enum SortBy {
+  RelevanceDesc = 'relevanceDesc',
+  NameAsc = 'nameAsc',
+  NameDesc = 'nameDesc',
+}
+
+export const defaultSortBy = SortBy.RelevanceDesc;
+
+const sortMapping = {
+  [SortBy.RelevanceDesc]: {
+    sortBy: SortBySearchOption.Relevance,
+    sortOrder: SortOrder.Descending,
+  },
+  [SortBy.NameAsc]: {
+    sortBy: SortBySearchOption.Name,
+    sortOrder: SortOrder.Ascending,
+  },
+  [SortBy.NameDesc]: {
+    sortBy: SortBySearchOption.Name,
+    sortOrder: SortOrder.Descending,
+  },
+};
+
+interface ToSearchParamsProps {
```

Comment:
You could omit the `To` here? The word only has meaning in a specific context (if you know a transformation occurred from something to something else), not outside of it?

Also, `ParamsProps` sounds a bit redundant :) Perhaps `SearchParams` could suffice?